#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Allow to build without existing libc-dev installed
# DP: Author: Daniel Jacobowitz
# DP: Upstream status: Pending (I think)
# DP: Date: 2003-10-28

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- glibc-2.3.2/Makerules.orig	2003-10-28 18:27:11.000000000 -0500
+++ glibc-2.3.2/Makerules	2003-10-28 18:27:31.000000000 -0500
@@ -479,6 +479,7 @@ ifeq (yes,$(elf))
 # not for shared objects
 $(common-objpfx)shlib.lds: $(common-objpfx)config.make $(..)Makerules
 	$(LINK.o) -shared -Wl,-O1 \
+		  -nostdlib -nostartfiles \
 		  $(sysdep-LDFLAGS) $(config-LDFLAGS) $(LDFLAGS.so) \
 		  -Wl,--verbose 2>&1 | \
 	  sed > $@T \
